airbyte-metrics/metrics-lib/src/main/java/io/airbyte/metrics/lib/OssMetricsRegistry.java

```diff
@@ -95,7 +95,19 @@ public enum OssMetricsRegistry implements MetricsRegistry {
       "oldest running job in seconds"),
   OVERALL_JOB_RUNTIME_IN_LAST_HOUR_BY_TERMINAL_STATE_SECS(MetricEmittingApps.METRICS_REPORTER,
       "overall_job_runtime_in_last_hour_by_terminal_state_secs",
-      "overall job runtime - scheduling and execution for all attempts - for jobs that reach terminal states in the last hour. tagged by terminal states.");
+      "overall job runtime - scheduling and execution for all attempts - for jobs that reach terminal states in the last hour. tagged by terminal states."),
+
+  TEMPORAL_WORKFLOW_RESTART_ATTEMPT(MetricEmittingApps.WORKER,
+      "temporal.workflow.restart.attempt",
+      "count of number of attempts to restart a workflow"),
+
+  TEMPORAL_WORKFLOW_RESTART_SUCCESS(MetricEmittingApps.WORKER,
+      "temporal.workflow.restart.success",
+      "count of number of successful restarts performed by the workflow connection manager."),
+
+  TEMPORAL_WORKFLOW_RESTART_FAILURE(MetricEmittingApps.WORKER,
+      "temporal.workflow.restart.failure",
+      "count of the number of workflow restart failures");
```

Comment:
Thinking out loud: 
These are all related to the sync workflow a.k.a the child workflow. 

1) Do we want to also track the connection manager workflow runs success/failure? I think it is and would love thoughts here.
2) I'm thinking we can make clearer what workflow these refer to, either via a name change, better descriptions, different tags etc. As is, I think this refers to all workflows when it only refers to temporal sync workflows.

Reply:
@davinchia I think that is the logical evolution of this (track syncs AND restarts).  The question becomes whether or not the metric should just be something like `temporal.workflow.(attempt|success|failure)` and have an attribute/tag of `type` (sync|restart value) or if the type should be part of the metric name.  I suppose the basic question to ask ourselves is whether or not we would want to see ALL workflows and/or be able to filter by type or if we will always just filter by type on a dashboard/alert.  We should answer this now to get the names correct, but we don't have to add the metrics for tracking the syncs as part of this PR -- it should probably be a separate PR in my opinion just so that it gets the proper focus.